From becbcd27e09030cc1032e013ff08c726acae79bf Mon Sep 17 00:00:00 2001
From: Andrei Vagin <avagin@google.com>
Date: Sun, 20 Mar 2022 22:13:45 -0700
Subject: [PATCH 227/249] scripts/ci: mount test cgroups once

zdtm.py mounts two named controllers for tests. In CI, we run zdtm.py a few
times, so we can mount (create) these controllers once to avoid any unwanted
effects.

Signed-off-by: Andrei Vagin <avagin@google.com>
---
 scripts/ci/run-ci-tests.sh | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/scripts/ci/run-ci-tests.sh b/scripts/ci/run-ci-tests.sh
index 22420b922..14e6b1224 100755
--- a/scripts/ci/run-ci-tests.sh
+++ b/scripts/ci/run-ci-tests.sh
@@ -139,6 +139,13 @@ time make unittest
 
 ulimit -c unlimited
 
+cgid=$$
+cleanup_cgroup() {
+	./test/zdtm_umount_cgroups $cgid
+}
+trap cleanup_cgroup EXIT
+./test/zdtm_mount_cgroups $cgid
+
 echo "|$(pwd)/test/abrt.sh %P %p %s %e" > /proc/sys/kernel/core_pattern
 
 if [ "${COMPAT_TEST}x" = "yx" ] ; then
-- 
2.35.1

